connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ExactlyOnceWorkerSourceTask.java

```diff
@@ -280,16 +280,18 @@ private void commitTransaction() {
 
         // Commit the transaction
         // Blocks until all outstanding records have been sent and ack'd
-        try {
-            producer.commitTransaction();
-        } catch (Throwable t) {
-            log.error("{} Failed to commit producer transaction", ExactlyOnceWorkerSourceTask.this, t);
-            flushError.compareAndSet(null, t);
+        Throwable error = flushError.get();
+        if (error == null) {
```

Comment:
Can this be the else block of https://github.com/apache/kafka/pull/12784/files#diff-6073f350e3c1c4ca7334407b32d162012543dad6bcc048cc970f3d11e852e5f2L293? Seems like `transactionOpen = false;` is anyways being set in this case. WDYT?

Reply:
want to check for https://github.com/apache/kafka/pull/12784/files#diff-6073f350e3c1c4ca7334407b32d162012543dad6bcc048cc970f3d11e852e5f2R274 first here. 
If this was null, next error could be result of producer.commitTransaction itself, so we need to check for that error again. 
Error resulting from any of those should throw exception... which is what i'm intending to do. 
lmk if i am missing your suggesting 😅

Reply:
Yeah it makes sense now. I missed the part on line #289.